crates/forge/bin/cmd/test/mod.rs

```diff
@@ -342,46 +299,54 @@ impl TestArgs {
             return Ok(TestOutcome::new(results, self.allow_failure));
         }
 
-        // Set up identifiers
+        // Set up trace identifiers.
         let known_contracts = runner.known_contracts.clone();
         let mut local_identifier = LocalTraceIdentifier::new(&known_contracts);
         let remote_chain_id = runner.evm_opts.get_remote_chain_id();
-        // Do not re-query etherscan for contracts that you've already queried today.
         let mut etherscan_identifier = EtherscanIdentifier::new(&config, remote_chain_id)?;
 
-        // Set up test reporter channel
+        // Run tests.
         let (tx, rx) = channel::<(String, SuiteResult)>();
-
-        // Run tests
         let handle = tokio::task::spawn({
             let filter = filter.clone();
             async move { runner.test(&filter, tx, test_options).await }
         });
 
-        let mut results = BTreeMap::new();
         let mut gas_report = GasReport::new(config.gas_reports, config.gas_reports_ignore);
-        let sig_identifier =
-            SignaturesIdentifier::new(Config::foundry_cache_dir(), config.offline)?;
 
-        let mut total_passed = 0;
-        let mut total_failed = 0;
-        let mut total_skipped = 0;
-        let mut suite_results: Vec<TestOutcome> = Vec::new();
+        // Build the trace decoder.
```

Comment:
- Multiple decoders don't achieve anything more than a single one.
- Hoisted from inside of the loop as the only thing that was changing was the labels.